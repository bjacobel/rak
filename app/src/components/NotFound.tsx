import { styled } from '@linaria/react';

import { center, openSans } from '../stylesheets/shared.css';
import StyledLink from './StyledLink';

const Header = styled.h1`
  ${center}
  ${openSans}
`;

export default () => (
  <div>
    <Header>404: page not found</Header>
    <StyledLink to="/">Home</StyledLink>
  </div>
);
