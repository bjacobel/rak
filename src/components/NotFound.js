import React, { Component } from 'react';
import { Link } from 'react-router-dom';

import styles from '../stylesheets/notFound.css';
import { link } from '../stylesheets/link.css';

export default class NotFound extends Component {
  render() {
    return (
      <div>
        <h1 className={styles['not-found']}>404: page not found</h1>
        <Link className={link} to="/">
          Home
        </Link>
      </div>
    );
  }
}
