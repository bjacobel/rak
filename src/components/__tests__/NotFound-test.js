import React from 'react';
import { shallow } from 'enzyme';

import NotFound from 'components/NotFound';

describe('404 component', () => {
  it('matches snapshot', () => {
    expect(shallow(<NotFound />)).toMatchSnapshot();
  });

  it('displays 404 message', () => {
    const notFound = shallow(<NotFound />);

    expect(notFound.find('h1').length).toBe(1);
    expect(notFound.find('h1').text()).toEqual('404: page not found');
  });

  it('has a link back to home', () => {
    const notFound = shallow(<NotFound />);
    const homeLink = notFound.find('Link');

    expect(homeLink.length).toBe(1);
    expect(homeLink.prop('to')).toEqual('/');
  });
});
