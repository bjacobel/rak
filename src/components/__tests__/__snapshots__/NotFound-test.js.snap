// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`404 component matches snapshot 1`] = `
<div>
  <h1>
    404: page not found
  </h1>
  <Link
    to="/"
  >
    Home
  </Link>
</div>
`;
